File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
//         remove trailing and leading whitespace
        s = s.trim();
        s = s.replaceAll("\\s+", " ");
        // System.out.println(s);
        
        List<String> wordsList = Arrays.asList(s.split(" "));
        Collections.reverse(wordsList);
        
        s = String.join(" ", wordsList);
        
        return s;
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let reversedArr = s.split(" ");
    reversedArr.reverse();
    let newArr = [];
    reversedArr.forEach(word => {
        if (word !== "") {
            newArr.push(word);
        }
    });
    return newArr.join(" ");
};